### PR TITLE
ARTEMIS-3050 Reduce PagedReferenceImpl memory footprint

### PR DESCRIPTION
**Pull request by franz1981**
https://issues.apache.org/jira/browse/ARTEMIS-3050

### CODE REVIEW

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/PagedReferenceImpl.java [hunk 1]
```diff
@@ -30,10 +30,9 @@
 import org.apache.activemq.artemis.core.server.ServerConsumer;
 import org.apache.activemq.artemis.core.server.impl.AckReason;
 import org.apache.activemq.artemis.core.transaction.Transaction;
-import org.apache.activemq.artemis.utils.collections.LinkedListImpl;
 import org.jboss.logging.Logger;
 
-public class PagedReferenceImpl extends LinkedListImpl.Node<PagedReferenceImpl> implements PagedReference, Runnable {
```
Comment (clebertsuconic) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/PagedReferenceImpl.java:36]: This was an optimization you performed some time ago, to speed up things on the delivering... Queue.references...

Are you sure about this change? Did you validate perf on it?

Comment (franz1981) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/PagedReferenceImpl.java:36]: Yes but i was young :)
I didn't know that pages references are never placed on the queue, hence they don't need to extends node

Comment (michaelandrepearce) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/PagedReferenceImpl.java:36]: <= He had hair back then and care free:P

**Comment by franz1981** [comment]
This one should be safe to be merged

**Comment by clebertsuconic** [comment]
lets postpone this PR.. and think about the WeakREference on PagedReference

perhaps this will be ok, as the messages will be sent through the LInkedList on Queue in certain occasions. I guess if we make the WeakReference on PagedReference a strong reference, it willl probably be ok.

**Comment by clebertsuconic** [comment]
postponing after the next release... as discussed.

**Comment by michaelandrepearce** [comment]
@clebertsuconic  / @franz1981  whats occuring with this, wanting to start clearing down old / stagnant PR's a bit of spring cleaning so to say - so we focus on stuff thats actively working on / relevant .

**Comment by franz1981** [comment]
@michaelandrepearce I was waiting the comment of @clebertsuconic 
```
postponing after the next release... as discussed.
```
So don't know yet what are the replacement/proposal to this yet

**Comment by jbertram** [comment]
I'm converting this to a draft and removing the `DO-NOT-MERGE-YET` label. Using the draft functionality is a bit more elegant. That said, it would be great to either get this merged or just close it. What do you think @clebertsuconic, @franz1981?

**Comment by jbertram** [comment]
@clebertsuconic, what do you think about this PR now that you've eliminated the soft cache? Should we merge it or close it?

**Pull request closed without merging**